### ISSUE TITLE
Ticket should be visible for more than one customer

### ISSUE THREAD

**Issue by gwingend** [zammad/zammad#1307]
### Infos:

* Used Zammad version: 1.6.1
* Used Zammad installation source: (source, package, ...)
* Operating system: 
* Browser + version: 

### Expected behavior:
In some conditions it would be useful that an additional customer could view a ticket.
So it would be cool if you could either assign a ticket to multiple customers or add a "viewer-field" to the ticket where additional customers who should be also able to view the ticket could be added.

usecases:
- two users are doing desksharing. User of the early shift reports the issue but it can´t be solved until the end of the shift. So the user of the late shift should see the ticket with his account and work with the agent on the solution. Next day the user of the morning shift should be able to view what happened after he went home the day before.
- in some conditions the boss wan´t to view what´s happening with a ticket his employee opened.
- the employee in financial department wants to get some special rights in the IT system. So the leader  of financial department opens a ticket for group usermanagement to give the employee the rights. Then it would be useful to add the affected employee to the ticket so boss and employee get the same information about ticket progress.

### Actual behavior:
not available.

**Comment by edneymatias** [comment]
Hi, organizations can be configured to share tickets between users participating in that organization. I think all you have to do is put all users that must share tickets in the same organization and set shared organization to 'yes'.

**Comment by hanneshal** [comment]
@edneymatias your absolut right. Company Tickets, using a shared organisation is the way to go. @gwingend could you test this please.

**Comment by gwingend** [comment]
Hi,

thanks for the suggestions.
I know the organization feature and I think this is pretty cool. :-)

But this is not working for my usecase.
The helpdesk stuff is responsible for the IT of the company not for external customers.

So if choosing the company to be the organisation and set the organisation to share the tickets, each of the more than 500 users could see the tickets of all users, no matter which department he belongs to.

This won´t be accepted due to security reasons and it wouldn´t be accepted by the staff council too.

In general no tickets should be shared. But in some individual cases the technican should decide if he needs to have someone "in cc" who is able to view the ticket.
(we have the feature in our old tickets system we wan´t to replace with zammad.


I thought about choosing the department the user belongs to as the organization. 
But every user has the same mail domain. 
So domain based membership to an organization won´t work and manually add each user to the organization (department) in zammad would lead to a lot manual administration. 

Maybe in ldap environments a Organization membership based on ldap attributes or ldap group memberships would be a nice feature anyway. 
So the agent could see all tickets in the department the user belongs to.

**Comment by edneymatias** [comment]
@gwingend ,

> two users are doing desk sharing. The user of the early shift reports the issue but it can´t be solved until the end of the shift. 

I think that the groups in Zammad do that by default, I mean they restrict access to tickets based on the agent group. I did a test here and users in the SAD group can't see tickets from users in the SAB group or maybe I did something wrong. You can assign a user to more than one group.

> in some conditions the boss want to view what´s happening with a ticket his employee opened.

In this case, you will have to assign the boss to every group he must track and build overviews that allow him to filter tickets by groups. Overviews can be assigned to roles and/or to specific users. So boss A can have different overviews of boss B.

> Then it would be useful to add the affected employee to the ticket so boss and employee get the same information about ticket progress.

I don't know how to assign third parties to tickets but it would be great to have the possibility to sign for ticket updates. From #208, it seems is on the way!

Best regards.

**Comment by gwingend** [comment]
@edneymatias,

> I think that the groups in Zammad do that by default, I mean they restrict access to tickets based on the agent group. I did a test here and users in the SAD group can't see tickets from users in the SAB group or maybe I did something wrong. You can assign a user to more than one group.


You are right. But you are talking about "users" in the means of "agents" or "technicans". 
When I´m talking about users I mean internal endusers that call the helpdesk. 
So "user" = "customer".

> In this case, you will have to assign the boss to every group he must track and build overviews that allow him to filter tickets by groups. Overviews can be assigned to roles and/or to specific users. So boss A can have different overviews of boss B.

Same here. You are talking about boss as boss of agents or technicans.
In my scenario boss means boss of the endusers.

> I don't know how to assign third parties to tickets but it would be great to have the possibility to sign for ticket updates. From #208, it seems is on the way!

Yes. What I mean is nearly the same as #208 but for customers instead of agents.
But the feature for agents in #208 is also really cool! :-)

**Comment by edneymatias** [comment]
Now I see what you meant, sorry for it taking so long!

**Comment by banholzer** [comment]
We would also appreciate such behaviour.

**Comment by hfischer-github** [comment]
Are there any news regaring that feature request? We would also be glad, if all users would have access to a ticket which are also part of email communication (to/cc). Often a user is creating a ticket via email and adds further user to email cc. All these users should have access to the ticket and the current status.

**Comment by MrGeneration** [comment]
This is a feature backlog.
This hasn't move in any direction - if things happen regarding this topic, we'll update this issue accordingly.

I'm locking the conversation of this issue to contributors only to reduce noise on this repository.
Please don't get me wrong, it's just that bumping issues doesn't speed up the pace. :(
